pandas/core/groupby/ops.py

```diff
@@ -575,6 +565,9 @@ def _cython_operation(
         elif is_datetimelike and kind == "aggregate":
             result = result.astype(orig_values.dtype)
 
+        if is_extension_array_dtype(orig_values.dtype):
```

Comment:
if you move line 568 (the is_extension_array_dtype) before line 558, can we then completely remove lines 558-556? (these are *already* extension dtypes).

Reply:
No, it looks like we still get a lot of failures